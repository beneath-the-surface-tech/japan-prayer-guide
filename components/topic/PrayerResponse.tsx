import { TFunction, Trans, useTranslation } from "next-i18next"
import React from "react"
import { Col, Container, FloatingLabel, Form, Row } from "react-bootstrap"
import Button from "react-bootstrap/Button"
import PrayerPoints, { PrayerDisplayStyle } from "../common/PrayerPoints"

const PrayerResponse = ({ topicTrans }: { topicTrans: TFunction }) => {
    const { t: common } = useTranslation("common")

    const textPlaceholder: string = common("response.feedbackTextPlaceholder", "")
    const emailPlaceholder: string = common("response.feedbackEmailPlaceholder", "")

    const topicVideoSrc: string = topicTrans("videoSrc", "")
    const couldLoadVideo: boolean = topicVideoSrc !== ""

    return (
        <Container fluid id="prayer-response" data-testid="prayer-response" className="pb-5">
            <Container>
                <h2 className="fw-bold text-primary">
                    <Trans t={common} i18nKey="response.title" />
                </h2>
                <p>
                    <Trans t={common} i18nKey="response.blurb0" />
                </p>
                <p>
                    <Trans t={common} i18nKey="response.blurb1" />
                </p>

                <h3 className="fw-bold fs-2">
                    <Trans t={common} i18nKey="response.recordingLabel" />
                </h3>
                <div className="mb-5 video-embed">
                    {couldLoadVideo ? (
                        <iframe
                            className="d-flex align-content-start"
                            src={topicVideoSrc}
                            allow="autoplay; fullscreen"
                            style={{
                                position: "absolute",
                                top: "0",
                                left: "0",
                                width: "100%",
                                height: "100%",
                                border: "0px",
                            }}
                            title="Prayer recording"
                        />
                    ) : (
                        <div>
                            <Trans t={common} i18nKey="errors.videoLoadError" />
                        </div>
                    )}
                </div>
                <Row>
                    <Col lg={6} md={12} className="px-0">
                        <PrayerPoints topicTrans={topicTrans} displayStyle={PrayerDisplayStyle.TopicBottom} />
                    </Col>
                    <Col lg={6} md={12}>
                        <Container data-testid="feedback" fluid className="px-0">
                            <h3 className="fw-bold mt-0">
                                <Trans t={common} i18nKey="response.feedbackLabel" />
                            </h3>
                            <Form action="https://formbold.com/s/67Gj1" method="POST">
                                <Form.Group className="mb-2">
                                    <FloatingLabel
                                        controlId="floatingTextArea"
                                        label={textPlaceholder}
                                        className="text-muted"
                                    >
                                        <Form.Control
                                            as="textarea"
                                            name="message"
                                            placeholder={textPlaceholder}
                                            style={{ height: "100px" }}
                                        />
                                    </FloatingLabel>
                                </Form.Group>
                                <Form.Group className="mb-3">
                                    <Form.Control type="email" placeholder={emailPlaceholder} />
                                </Form.Group>
                                <Button variant="outline-primary" className="fw-bold px-4 py-2" type="submit">
                                    <Trans t={common} i18nKey="response.feedbackSubmit" />
                                </Button>
                            </Form>
                        </Container>
                    </Col>
                </Row>
            </Container>
        </Container>
    )
}

export default PrayerResponse
