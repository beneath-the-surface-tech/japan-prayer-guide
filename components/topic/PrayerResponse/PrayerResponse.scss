// -------------------------
// Prayer Response
// -------------------------

#prayer-response {
    background-color: $grey-2;
    p {
        font-size: 20px;
        @include media-breakpoint-down(md) {
            font-size: 18px;
        }
    }

    h2 {
        font-family: $font-family-secondary;
        padding-top: 4rem;
        @include media-breakpoint-down(md) {
            padding-top: 2.5rem;
        }
    }

    h3 {
        margin-top: 2rem;
        font-family: $font-family-secondary;
        line-height: normal;
    }

    #topic-prayer-topicBottom {
        padding-left: 0;
        padding-right: 0;
    }

    .video-embed {
        position: relative;
        padding-bottom: 56.25% !important; /* 16:9 */
        margin-bottom: 30px;
        height: 0;
    }

    .reflections-submission-heading {
        font-size: 24px;
        @include media-breakpoint-down(sm) {
            font-size: 22px;
        }
        font-family: $font-family-secondary;
        font-weight: 700;
    }
}
