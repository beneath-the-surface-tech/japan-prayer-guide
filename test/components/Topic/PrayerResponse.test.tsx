import { render, screen } from "@testing-library/react"
import PrayerResponse from "../../../components/topic/PrayerResponse"

jest.mock("next-i18next", () => ({
    // this mock makes sure any components using the translate hook can use it without a warning being shown
    useTranslation: () => {
        return {
            t: (str: any) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        }
    },
    Trans: (props: any) => {
        if (props.children) {
            // mock for <Trans>{value}</Trans>
            return props.children
        } else if (props.i18nKey) {
            // mock for <Trans t={t} i18nKey={value} />
            return props.i18nKey
        }
        return props
    },
}))

jest.mock("../../../components/common/PrayerPoints")

describe("Responding in Prayer", () => {
    beforeEach(() => {
        const useRouter = jest.spyOn(require("next/router"), "useRouter")
        useRouter.mockReturnValue({
            push: () => {},
        })
    })

    test("Snapshot test", () => {
        // This test is to check against unintended changes.
        // If the change is intentional you can update the snapshot with `jest --updateSnapshot`

        const component = render(<PrayerResponse topicTrans={jest.fn()} />)
        expect(component).toMatchSnapshot()
    })

    test("Renders a section with the right text inside", () => {
        render(<PrayerResponse topicTrans={jest.fn()} />)
        const responseSection = screen.getByTestId("prayer-response")
        const responseTitle = responseSection.querySelector("h2")

        const feedbackSection = screen.getByTestId("feedback")
        const feedbackTitle = responseSection.querySelector("h3")
        const feedbackLabel = feedbackSection.querySelector("textarea[name='message']")
        const feedbackEmail: HTMLInputElement = feedbackSection.querySelector('input[type="email"]') as HTMLInputElement
        const feedbackSubmit = feedbackSection.querySelector('button[type="submit"]')

        expect(responseTitle?.textContent).toBe("response.title")
        expect(responseTitle).toHaveClass("fw-bold", "text-primary")

        expect(feedbackTitle?.textContent).toBe("response.recordingLabel")
        expect(feedbackLabel?.getAttribute("placeholder")).toBe("response.feedbackTextPlaceholder")
        expect(feedbackEmail?.type).toBe("email")

        expect(feedbackSubmit).toHaveClass("fw-bold", "px-4", "py-2")
        expect(feedbackSubmit?.textContent).toBe("response.feedbackSubmit")
    })
})
