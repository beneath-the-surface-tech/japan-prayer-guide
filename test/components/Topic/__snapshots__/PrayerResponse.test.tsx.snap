// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Responding in Prayer Snapshot test 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pb-5 container-fluid"
        data-testid="prayer-response"
        id="prayer-response"
      >
        <div
          class="container"
        >
          <h2
            class="fw-bold text-primary"
          >
            response.title
          </h2>
          <p>
            response.blurb
          </p>
          <h3
            class="fw-bold fs-2"
          >
            response.recordingLabel
          </h3>
          <div
            class="row"
          >
            <div
              class="px-0 col-lg-6 col-md-12"
            />
            <div
              class="col-lg-6 col-md-12"
            >
              <div
                class="px-0 container-fluid"
                data-testid="feedback"
              >
                <h3
                  class="fw-bold mt-0"
                >
                  response.feedbackLabel
                </h3>
                <form
                  action="https://formbold.com/s/3Zgzd"
                  class=""
                  method="POST"
                >
                  <div
                    class="mb-2"
                  >
                    <div
                      class="text-muted form-floating"
                    >
                      <textarea
                        class="form-control"
                        id="floatingTextArea"
                        name="message"
                        placeholder="response.feedbackTextPlaceholder"
                        style="height: 100px;"
                      />
                      <label
                        for="floatingTextArea"
                      >
                        response.feedbackTextPlaceholder
                      </label>
                    </div>
                  </div>
                  <div
                    class="mb-3"
                  >
                    <input
                      class="form-control"
                      placeholder="response.feedbackEmailPlaceholder"
                      type="email"
                    />
                  </div>
                  <button
                    class="fw-bold px-4 py-2 btn btn-outline-primary"
                    type="submit"
                  >
                    response.feedbackSubmit
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pb-5 container-fluid"
      data-testid="prayer-response"
      id="prayer-response"
    >
      <div
        class="container"
      >
        <h2
          class="fw-bold text-primary"
        >
          response.title
        </h2>
        <p>
          response.blurb
        </p>
        <h3
          class="fw-bold fs-2"
        >
          response.recordingLabel
        </h3>
        <div
          class="row"
        >
          <div
            class="px-0 col-lg-6 col-md-12"
          />
          <div
            class="col-lg-6 col-md-12"
          >
            <div
              class="px-0 container-fluid"
              data-testid="feedback"
            >
              <h3
                class="fw-bold mt-0"
              >
                response.feedbackLabel
              </h3>
              <form
                action="https://formbold.com/s/3Zgzd"
                class=""
                method="POST"
              >
                <div
                  class="mb-2"
                >
                  <div
                    class="text-muted form-floating"
                  >
                    <textarea
                      class="form-control"
                      id="floatingTextArea"
                      name="message"
                      placeholder="response.feedbackTextPlaceholder"
                      style="height: 100px;"
                    />
                    <label
                      for="floatingTextArea"
                    >
                      response.feedbackTextPlaceholder
                    </label>
                  </div>
                </div>
                <div
                  class="mb-3"
                >
                  <input
                    class="form-control"
                    placeholder="response.feedbackEmailPlaceholder"
                    type="email"
                  />
                </div>
                <button
                  class="fw-bold px-4 py-2 btn btn-outline-primary"
                  type="submit"
                >
                  response.feedbackSubmit
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
